pkg/api/stream.go

```diff
@@ -146,6 +146,10 @@ type Sink interface {
 	Closable
 }
 
+type PropsValidator interface {
```

Comment:
As the experimental interface, shall we move these and ping interface to internal package. So that we can easily remove them for future changes.